In [22]:
import numpy as np 
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error,mean_absolute_error
from sklearn.model_selection import train_test_split

In [37]:
#Building model
X=pd.read_csv(r"C:\Users\Dipika\Downloads\Linear_X_Train.csv")
Y=pd.read_csv(r"C:\Users\Dipika\Downloads\Linear_Y_Train.csv")
X_train,X_test,Y_train,Y_test=train_test_split(X,Y,test_size=0.4,random_state=10)
print(X.isnull().sum(),Y.isnull().sum())

x    0
dtype: int64 y    0
dtype: int64


In [38]:
#Validating LinearRegression
model=LinearRegression().fit(X_train,Y_train)
pred_y=model.predict(X_test)
training_accuracy=model.score(X_train,Y_train)
testing_accuracy=model.score(X_test,Y_test)
print("training_accuracy :",training_accuracy)
print("testing_accuracy :",testing_accuracy)
print("error: ",mean_absolute_error(Y_test,pred_y))

training_accuracy : 0.9713461243784723
testing_accuracy : 0.97035607922334
error:  11.102246606730363


Conclusion , As score is above 95% that means this means LinearRegression gives overfit model though error is less

In [39]:
from sklearn.tree import DecisionTreeRegressor
def model(method):
    model=DecisionTreeRegressor(criterion=method).fit(X_train,Y_train)
    pred_y=model.predict(X_test)
    training_accuracy=model.score(X_train,Y_train)
    testing_accuracy=model.score(X_test,Y_test)
    print("training_accuracy :",training_accuracy)
    print("testing_accuracy :",testing_accuracy)
    print("error: ",mean_absolute_error(Y_test,pred_y))

In [40]:
model('mse')

training_accuracy : 1.0
testing_accuracy : 0.9399629103335192
error:  15.752918140010001


In [48]:
from sklearn import svm
model=svm.SVR(kernel='rbf',C=1).fit(X_train,Y_train.values.ravel())
pred_y=model.predict(X_test)
training_accuracy=model.score(X_train,Y_train)
testing_accuracy=model.score(X_test,Y_test)
print("training_accuracy :",training_accuracy)
print("testing_accuracy :",testing_accuracy)
print("error: ",mean_absolute_error(Y_test,pred_y))

training_accuracy : 0.8781968772011938
testing_accuracy : 0.8784625867410017
error:  16.052455890665005


In [49]:
from sklearn.ensemble import RandomForestRegressor,GradientBoostingRegressor,BaggingRegressor

In [53]:
model=RandomForestRegressor(n_estimators=1).fit(X_train,Y_train.values.ravel())
pred_y=model.predict(X_test)
training_accuracy=model.score(X_train,Y_train)
testing_accuracy=model.score(X_test,Y_test)
print("training_accuracy :",training_accuracy)
print("testing_accuracy :",testing_accuracy)
print("error: ",mean_absolute_error(Y_test,pred_y))

training_accuracy : 0.9805142294666102
testing_accuracy : 0.9437139257577082
error:  15.272892804476665


In [59]:
model=GradientBoostingRegressor(n_estimators=17).fit(X_train,Y_train.values.ravel())
pred_y=model.predict(X_test)
training_accuracy=model.score(X_train,Y_train)
testing_accuracy=model.score(X_test,Y_test)
print("training_accuracy :",training_accuracy)
print("testing_accuracy :",testing_accuracy)
print("error: ",mean_absolute_error(Y_test,pred_y))

training_accuracy : 0.9409366072783077
testing_accuracy : 0.9345156706294874
error:  15.9475592767673


In [61]:
model=BaggingRegressor(n_estimators=1).fit(X_train,Y_train.values.ravel())
pred_y=model.predict(X_test)
training_accuracy=model.score(X_train,Y_train)
testing_accuracy=model.score(X_test,Y_test)
print("training_accuracy :",training_accuracy)
print("testing_accuracy :",testing_accuracy)
print("error: ",mean_absolute_error(Y_test,pred_y))

training_accuracy : 0.9794565225762604
testing_accuracy : 0.9410474981008314
error:  15.621486357580666


#### Conclusion : 
1.As GradientBoostingRegressor gives training_accuracy : 0.9409366072783077, testing_accuracy : 0.9345156706294874,
error:  15.9475592767673 with n =17 that means this is best fit model with less error, we accept this model
2. As Svm gives training_accuracy : 0.8781968772011938,testing_accuracy : 0.8784625867410017,
error:  16.052455890665005 with kernel ='rbf'that means this model we can also accept as best fit model.

SyntaxError: invalid syntax (<ipython-input-62-f188d5ac1197>, line 1)